docs/dev/dev-proxy/how-to/create-custom-plugin.md

```diff
@@ -0,0 +1,160 @@
+author: eliostruyf
+ms.author: 
+ms.date: 03/28/2024
+---
+
+# Create a custom plugin
+
+In this article, you will learn how to create a custom plugin for the Dev Proxy. By creating plugins for Dev Proxy, you can extend its functionality and add custom features to fit your needs.
+
+## Prerequisites
+
+Before you start creating a custom plugin, make sure you have the following prerequisites:
+
+- [.NET Core SDK](https://dotnet.microsoft.com/download)
+- [Visual Studio Code](https://code.visualstudio.com/)
+- The latest version of the Dev Proxy Abstractions DLL, which you can find on the [Dev Proxy GitHub releases](https://github.com/microsoft/dev-proxy/releases) page
+
+## Create a new plugin
+
+Follow the next steps to create a new project:
+
+1. Create a new class library project using the `dotnet new classlib` command.
+
+    ```bash
+    dotnet new classlib -n MyCustomPlugin
+    ```
+
+1. Open the newly created project in Visual Studio Code.
+
+    ```bash
+    code MyCustomPlugin
+    ```
+
+1. Add the Dev Proxy Abstractions DLL (`dev-proxy-abstractions.dll`) to the project folder.
+1. Add the `dev-proxy-abstractions.dll` as a reference to your project `DevProxyCustomPlugin.csproj` file.
+
+    ```xml
+    <ItemGroup>
+      <Reference Include="dev-proxy-abstractions">
+        <HintPath>.\dev-proxy-abstractions.dll</HintPath>
+        <Private>false</Private>
+        <ExcludeAssets>runtime</ExcludeAssets>
+      </Reference>
+    </ItemGroup>
+    ```
+
+1. Add the NuGet packages required for your project.
+
+    ```bash
+    dotnet add package Microsoft.Extensions.Configuration
+    dotnet add package Microsoft.Extensions.Configuration.Binder
+    dotnet add package Microsoft.Extensions.Logging.Abstractions
+    dotnet add package Titanium.Web.Proxy
+    ```
+
+1. Exclude the dependency DLLs from the build output by adding a `ExcludeAssets` tag per `PackageReference` in the `DevProxyCustomPlugin.csproj` file.
+
+    ```xml
+    <ExcludeAssets>runtime</ExcludeAssets>
+    ```
+
+1. Create a new class that implements the `BaseProxyPlugin` interface.
+
+    ```csharp
+    using Microsoft.DevProxy.Abstractions;
+    using Microsoft.Extensions.Configuration;
+
+    namespace MyCustomPlugin;
+
+    public class CatchApiCallsConfiguration
+    {
+      public string? RequiredHeader { get; set; }
+    }
+
+    public class CatchApiCalls : BaseProxyPlugin
+    {
+      public override string Name => nameof(CatchApiCalls);
+
+      private readonly CatchApiCallsConfiguration _configuration = new();
+
+      public override void Register(IPluginEvents pluginEvents,
+                                    IProxyContext context,
+                                    ISet<UrlToWatch> urlsToWatch,
+                                    IConfigurationSection? configSection = null)
+      {
+        base.Register(pluginEvents, context, urlsToWatch, configSection);
+
+        // Bind your plugin configuration
+        configSection?.Bind(_configuration);
```

Comment:
This is optional. Some plugins might have no configuration

Reply:
I split it; the first code sample is without the configuration, and the next one with the configuration is highlighted as optional.